# AirBnB NY Locations Data Case Study

In this final project, you task will be to take the data provided and find evidance to answer the following questions.

1. Which hosts are the busiest and why?
2. How many neighborhood groups are available and which shows up the most?
3. Are private rooms the most popular in manhattan?
4. Which hosts are the busiest and based on their reviews?
5. Which neighorhood group has the highest average price?
6. Which neighborhood group has the highest total price?
7. Which top 5 hosts have the highest total price?
8. Who currently has no (zero) availability with a review count of 100 or more?
9. What host has the highest total of prices and where are they located?
10. When did Danielle from Queens last receive a review?

You will be given **4 hours** to complete this assignment. 
**Be Advised** I will go dark for this intire assignment time period. That said, any questions that you would like to ask about the data, or the project **MUST** be asked before the time starts. Once the time has started, I can no longer give information.

This is to similate what you will face when you are out in the wild. 

Happy Coding!

In [2]:
import numpy as np
import pandas as pd
import matplotlib.pyplot as plt

In [3]:
air_bnb = pd.read_csv('AB_NYC_2019.csv')
air_bnb.head()

id                                              name  host_id  \
0  2539                Clean & quiet apt home by the park     2787   
1  2595                             Skylit Midtown Castle     2845   
2  3647               THE VILLAGE OF HARLEM....NEW YORK !     4632   
3  3831                   Cozy Entire Floor of Brownstone     4869   
4  5022  Entire Apt: Spacious Studio/Loft by central park     7192   

     host_name neighbourhood_group neighbourhood  latitude  longitude  \
0         John            Brooklyn    Kensington  40.64749  -73.97237   
1     Jennifer           Manhattan       Midtown  40.75362  -73.98377   
2    Elisabeth           Manhattan        Harlem  40.80902  -73.94190   
3  LisaRoxanne            Brooklyn  Clinton Hill  40.68514  -73.95976   
4        Laura           Manhattan   East Harlem  40.79851  -73.94399   

         room_type  price  minimum_nights  number_of_reviews last_review  \
0     Private room    149               1                  9  2018-10-19   
1  Entire home/apt    225               1                 45  2019-05-21   
2     Private room    150               3                  0         NaN   
3  Entire home/apt     89               1                270  2019-07-05   
4  Entire home/apt     80              10                  9  2018-11-19   

   reviews_per_month  calculated_host_listings_count  availability_365  
0               0.21                               6               365  
1               0.38                               2               355  
2                NaN                               1               365  
3               4.64                               1               194  
4               0.10                               1                 0

In [4]:
# How many neighborhood groups are available and which shows up the most?
n_groups = air_bnb['neighbourhood_group']
print(n_groups.value_counts())

print(f"There are {n_groups.value_counts().count()} groups, {n_groups.value_counts().keys()[0]} is the most frequent")

Manhattan        21661
Brooklyn         20104
Queens            5666
Bronx             1091
Staten Island      373
Name: neighbourhood_group, dtype: int64
There are 5 groups, Manhattan is the most frequent


In [5]:
# Are private rooms the most popular in manhattan?
r_type = air_bnb['room_type']
display(r_type.value_counts())

m_pop = air_bnb[air_bnb['neighbourhood_group']=='Manhattan']
m_pop

test = m_pop['room_type']
display(test.value_counts())

print(f"The most popular room type is: {test.value_counts().keys()[0]}, so false")

Entire home/apt    25409
Private room       22326
Shared room         1160
Name: room_type, dtype: int64

Entire home/apt    13199
Private room        7982
Shared room          480
Name: room_type, dtype: int64

The most popular room type is: Entire home/apt, so false


In [48]:
# Which hosts are the busiest and based on their reviews?

air_bnb.groupby('host_name').sum().sort_values('number_of_reviews', ascending = False).head()

id      host_id     latitude    longitude  price  \
host_name                                                             
Michael    7430617239  22673153604  16984.89137 -30841.28573  66895   
David      7943862898  28633613354  16414.34392 -29804.53757  65844   
John       5737372154  24525297065  11974.02914 -21743.58993  41892   
Jason      2176696732   6962647849   5700.22907 -10354.60535  19381   
Alex       5496620312  19850092761  11361.10533 -20635.86235  52563   

           minimum_nights  number_of_reviews  reviews_per_month  \
host_name                                                         
Michael              4600              11081             475.82   
David                3754               8103             508.61   
John                 2603               7223             321.02   
Jason                1193               6522             196.15   
Alex                 1651               6204             443.44   

           calculated_host_listings_count  availability_365  
host_name                                                    
Michael                              1043             38888  
David                                 907             44171  
John                                 2792             40249  
Jason                                 312             19338  
Alex                                  475             30031

In [46]:
#Which neighorhood group has the highest average price?
print('Manhattan')

air_bnb.groupby('neighbourhood_group').mean().sort_values('price', ascending = False).head()

Manhattan


id       host_id   latitude  longitude  \
neighbourhood_group                                                     
Manhattan            1.877494e+07  6.783062e+07  40.765062 -73.974607   
Brooklyn             1.825685e+07  5.671526e+07  40.685036 -73.951190   
Staten Island        2.159747e+07  9.853360e+07  40.610225 -74.105805   
Queens               2.175500e+07  9.615680e+07  40.731531 -73.872775   
Bronx                2.273492e+07  1.056099e+08  40.848305 -73.884552   

                          price  minimum_nights  number_of_reviews  \
neighbourhood_group                                                  
Manhattan            196.875814        8.579151          20.985596   
Brooklyn             124.383207        6.056556          24.202845   
Staten Island        114.812332        4.831099          30.941019   
Queens                99.517649        5.181433          27.700318   
Bronx                 87.496792        4.560953          26.004583   

                     reviews_per_month  calculated_host_listings_count  \
neighbourhood_group                                                      
Manhattan                     1.272131                       12.791330   
Brooklyn                      1.283212                        2.284371   
Staten Island                 1.872580                        2.319035   
Queens                        1.941200                        4.060184   
Bronx                         1.837831                        2.233731   

                     availability_365  
neighbourhood_group                    
Manhattan                  111.979410  
Brooklyn                   100.232292  
Staten Island              199.678284  
Queens                     144.451818  
Bronx                      165.758937

In [43]:
# Which neighbor hood group has the highest total price?

air_bnb.groupby('neighbourhood_group').sum().sort_values('price', ascending = False).head()

id        host_id      latitude     longitude  \
neighbourhood_group                                                            
Manhattan            406683958152  1469279004836  883012.01680 -1.602364e+06   
Brooklyn             367035792483  1140203577483  817931.96821 -1.486715e+06   
Queens               123263814259   544824426112  230784.85597 -4.185631e+05   
Bronx                 24803796188   115220380205   44565.50071 -8.060805e+04   
Staten Island          8055857451    36753031930   15147.61383 -2.764147e+04   

                       price  minimum_nights  number_of_reviews  \
neighbourhood_group                                               
Manhattan            4264527          185833             454569   
Brooklyn             2500600          121761             486574   
Queens                563867           29358             156950   
Bronx                  95459            4976              28371   
Staten Island          42825            1802              11541   

                     reviews_per_month  calculated_host_listings_count  \
neighbourhood_group                                                      
Manhattan                     21158.08                          277073   
Brooklyn                      21104.98                           45925   
Queens                         8879.05                           23005   
Bronx                          1609.94                            2437   
Staten Island                   587.99                             865   

                     availability_365  
neighbourhood_group                    
Manhattan                     2425586  
Brooklyn                      2015070  
Queens                         818464  
Bronx                          180843  
Staten Island                   74480

In [41]:
#Which top 5 hosts have the highest total price?
air_bnb.groupby('host_name').sum().sort_values('price', ascending = False).head()

id      host_id     latitude    longitude  price  \
host_name                                                                 
Sonder (NYC)  10885561678  71782340547  13316.25823 -24198.18856  82795   
Blueground     7210036953  24924786136   9451.60418 -17166.13165  70331   
Michael        7430617239  22673153604  16984.89137 -30841.28573  66895   
David          7943862898  28633613354  16414.34392 -29804.53757  65844   
Alex           5496620312  19850092761  11361.10533 -20635.86235  52563   

              minimum_nights  number_of_reviews  reviews_per_month  \
host_name                                                            
Sonder (NYC)            4353               1281             397.56   
Blueground              7470                 29               6.04   
Michael                 4600              11081             475.82   
David                   3754               8103             508.61   
Alex                    1651               6204             443.44   

              calculated_host_listings_count  availability_365  
host_name                                                       
Sonder (NYC)                          106929             98588  
Blueground                             53824             58884  
Michael                                 1043             38888  
David                                    907             44171  
Alex                                     475             30031

In [106]:
# Who currently has no (zero) availability with a review count of 100 or more?
air_bnb[air_bnb.number_of_reviews > 100].groupby('host_name').sum().sort_values('availability_365',ascending = True).head(20)

id    host_id  latitude  longitude  price  \
host_name                                                         
Ivy               140425     683975  40.67550  -73.95878     79   
Karin             286838    1492339  40.80748  -73.95589     75   
Coral              44221     193722  40.70666  -74.01374    196   
Gabriela        31570350  204160906  81.38813 -147.89156    103   
Mohamed         35088278  188200086  81.53406 -147.90847     70   
Misty             635282    1215949  40.71657  -73.96264    165   
Felipe          15080994   95419310  40.76276  -73.98942    140   
Andy & Friends   2547556    7143642  81.38203 -147.91275     70   
Véronique        2339096   11947308  40.71590  -73.99059     70   
Kent             3540370   17820079  40.72673  -73.98004    130   
Taylor & Tee     5590966   28723165  40.82778  -73.94884     70   
Terri            1546945    4983320  40.75431  -73.83131     59   
Micah            6000785   31150792  40.69709  -73.93538    100   
Dante           12628500   67261284  40.75123  -73.87993     67   
Krysta          10365373    8049682  40.67433  -73.95750     81   
Megan E         18505594   31658624  40.66998  -73.93960     44   
Masha            4628603    6465286  40.71191  -73.96376    180   
Lane             1734724    8910286  40.72076  -73.95614    180   
Dragan            531091    1787284  40.75532  -73.91603    109   
Long             8365647   10024006  40.88837  -73.90939     65   

                minimum_nights  number_of_reviews  reviews_per_month  \
host_name                                                              
Ivy                          2                115               1.18   
Karin                        2                101               1.10   
Coral                        3                114               1.06   
Gabriela                     2                319               9.81   
Mohamed                      2                233               8.27   
Misty                        2                179               2.20   
Felipe                       2                105               3.09   
Andy & Friends              10                230               3.15   
Véronique                    1                196               3.03   
Kent                         1                191               3.17   
Taylor & Tee                 1                131               2.50   
Terri                        1                116               1.63   
Micah                        2                143               3.41   
Dante                        1                204               5.27   
Krysta                       3                120               2.81   
Megan E                      3                104               4.00   
Masha                        3                103               1.92   
Lane                        30                183               2.63   
Dragan                      30                304               3.70   
Long                         1                202               4.35   

                calculated_host_listings_count  availability_365  
host_name                                                         
Ivy                                          1                 0  
Karin                                        1                 0  
Coral                                        1                 0  
Gabriela                                     4                 0  
Mohamed                                      4                 0  
Misty                                        1                 0  
Felipe                                       1                 0  
Andy & Friends                               8                 0  
Véronique                                    1                 0  
Kent                                         1                 0  
Taylor & Tee                                 1                 0  
Terri                                        2                 0  
Micah                   

In [113]:
# What host has the highest total of prices and where are they located?
air_bnb.groupby('host_name').sum().sort_values('price', ascending = False).head()


id      host_id     latitude    longitude  price  \
host_name                                                                 
Sonder (NYC)  10885561678  71782340547  13316.25823 -24198.18856  82795   
Blueground     7210036953  24924786136   9451.60418 -17166.13165  70331   
Michael        7430617239  22673153604  16984.89137 -30841.28573  66895   
David          7943862898  28633613354  16414.34392 -29804.53757  65844   
Alex           5496620312  19850092761  11361.10533 -20635.86235  52563   

              minimum_nights  number_of_reviews  reviews_per_month  \
host_name                                                            
Sonder (NYC)            4353               1281             397.56   
Blueground              7470                 29               6.04   
Michael                 4600              11081             475.82   
David                   3754               8103             508.61   
Alex                    1651               6204             443.44   

              calculated_host_listings_count  availability_365  
host_name                                                       
Sonder (NYC)                          106929             98588  
Blueground                             53824             58884  
Michael                                 1043             38888  
David                                    907             44171  
Alex                                     475             30031

In [120]:
# When did Danielle from Queens last receive a review?
air_bnb[air_bnb.host_name == 'Danielle'][air_bnb.neighbourhood_group == 'Queens'].sort_values('last_review', ascending = True)


C:\Users\thoma\AppData\Local\Temp/ipykernel_27964/1350018990.py:2: UserWarning: Boolean Series key will be reindexed to match DataFrame index.
  air_bnb[air_bnb.host_name == 'Danielle'][air_bnb.neighbourhood_group == 'Queens'].sort_values('last_review', ascending = True)


id                                               name    host_id  \
27021  21386105  Quiet & clean 1br haven with balcony near the ...  154256662   
33861  26814763  One bedroom with full bed / 1 stop from Manhattan  201647469   
7086    5115372  Comfy Room Family Home LGA Airport NO CLEANING...   26432133   
20403  16276632  Cozy Room Family Home LGA Airport NO CLEANING FEE   26432133   
22068  17754072                Bed in Family Home Near LGA Airport   26432133   
21517  17222454   Sun Room Family Home LGA Airport NO CLEANING FEE   26432133   
22469  18173787  Cute Tiny Room Family Home by LGA NO CLEANING FEE   26432133   
16349  13151075                    ASTORIA APARTMENT OUTDOOR SPACE   18051286   

      host_name neighbourhood_group     neighbourhood  latitude  longitude  \
27021  Danielle              Queens           Astoria  40.77134  -73.92424   
33861  Danielle              Queens  Long Island City  40.74565  -73.94699   
7086   Danielle              Queens     East Elmhurst  40.76374  -73.87103   
20403  Danielle              Queens     East Elmhurst  40.76335  -73.87007   
22068  Danielle              Queens     East Elmhurst  40.76389  -73.87155   
21517  Danielle              Queens     East Elmhurst  40.76367  -73.87088   
22469  Danielle              Queens     East Elmhurst  40.76380  -73.87238   
16349  Danielle              Queens           Astoria  40.77221  -73.92901   

             room_type  price  minimum_nights  number_of_reviews last_review  \
27021  Entire home/apt    250               3                  1  2018-01-02   
33861     Private room    108               2                 13  2019-06-20   
7086      Private room     54               1                430  2019-07-03   
20403     Private room     48               1                510  2019-07-06   
22068      Shared room     38               1                224  2019-07-06   
21517     Private room     48               1                417  2019-07-07   
22469     Private room     48               1                436  2019-07-08   
16349     Private room     50               1                  0         NaN   

       reviews_per_month  calculated_host_listings_count  availability_365  
27021               0.05                               1               180  
33861               1.74                               1               333  
7086               13.45                               5               347  
20403              16.22                               5               341  
22068               7.96                               5                80  
21517              14.36                               5               338  
22469              16.03                               5               337  
16349                NaN                               1                 0

## Further Questions

1. Which host has the most listings?

In [124]:
display(air_bnb.groupby('host_name').sum().sort_values('calculated_host_listings_count',ascending = False))
print('Sonder')

id      host_id     latitude    longitude  price  \
host_name                                                                 
Sonder (NYC)  10885561678  71782340547  13316.25823 -24198.18856  82795   
Blueground     7210036953  24924786136   9451.60418 -17166.13165  70331   
Kara           1981276355   4544361074   5827.31420 -10579.90583  36723   
Kazuya         3121544245  14147963198   4197.55601  -7612.96008   4514   
Sonder         2517132868   1175332896   3909.55849  -7103.58856  20451   
...                   ...          ...          ...          ...    ...   
Jerbean            819956      9269794     40.68615    -73.93095     75   
Jerald           14784557       767764     40.76688    -73.99402    200   
Jeonghoon        31089176     12625165     40.69160    -73.95961     42   
Jeny             20888611    150038776     40.76462    -73.93190     35   
현선               26139678     18497228     40.80485    -73.96534   1200   

              minimum_nights  number_of_reviews  reviews_per_month  \
host_name                                                            
Sonder (NYC)            4353               1281             397.56   
Blueground              7470                 29               6.04   
Kara                    3839                324              14.82   
Kazuya                  3090                 87              22.68   
Sonder                  2784                 43               8.65   
...                      ...                ...                ...   
Jerbean                    2                 38               0.51   
Jerald                     3                 31               0.94   
Jeonghoon                  1                  0               0.00   
Jeny                       1                 16               1.37   
현선                        60                  0               0.00   

              calculated_host_listings_count  availability_365  
host_name                                                       
Sonder (NYC)                          106929             98588  
Blueground                             53824             58884  
Kara                                   14679             39273  
Kazuya                                 10609             20237  
Sonder                                  9216             26555  
...                                      ...               ...  
Jerbean                                    1               365  
Jerald                                     1                51  
Jeonghoon                                  1                 0  
Jeny                                       1                95  
현선                                         1               365  

[11452 rows x 10 columns]

Sonder


2. How many listings have completely open availability?

In [137]:
sum(air_bnb['availability_365'] == 0)

17533

3. What room_types have the highest review numbers?

In [139]:
air_bnb.groupby('room_type').sum().sort_values('number_of_reviews', ascending = False).head()

id        host_id      latitude     longitude  \
room_type                                                                  
Entire home/apt  468495600287  1569156532318  1.034874e+06 -1.879267e+06   
Private room     434663231749  1618079882718  9.093203e+05 -1.650850e+06   
Shared room       26684386497   119044005530  4.724740e+04 -8.577428e+04   

                   price  minimum_nights  number_of_reviews  \
room_type                                                     
Entire home/apt  5381480          216152             580403   
Private room     2004450          120067             538346   
Shared room        81348            7511              19256   

                 reviews_per_month  calculated_host_listings_count  \
room_type                                                            
Entire home/apt           26565.34                          271834   
Private room              25529.62                           72062   
Shared room                1245.08                            5409   

                 availability_365  
room_type                          
Entire home/apt           2843783  
Private room              2482739  
Shared room                187921

# Final Conclusion

In this cell, write your final conclusion for each of the questions asked.

Also, if you uncovered some more details that were not asked above, please discribe them here.

-- Add your conclusion --

# Which hosts are the busiest and why?
How many neighborhood groups are available and which shows up the most? - 5, Manhattan
Are private rooms the most popular in manhattan? - No
Which hosts are the busiest and based on their reviews? - Michael 11,081
Which neighorhood group has the highest average price? - Manhattan 196.875814	
Which neighborhood group has the highest total price? - Manhattan 4264527
Which top 5 hosts have the highest total price? - Sonder, Blueground, Michael, David, Alex
Who currently has no (zero) availability with a review count of 100 or more? - Alot of peeps
What host has the highest total of prices and where are they located? - Sonder NYC
When did Danielle from Queens last receive a review? -2019 07 08